src/daemon/device_bragi.c

```diff
@@ -281,3 +283,69 @@ void bragi_delay(usbdevice* kb, delay_type_t type){
     // Since we use usbrecv for everything, the devices tell us when they are ready to handle another packet.
     return;
 }
+
+static void bragi_generate_encryption_key(uchar* encryption_key) {
+  ckb_generate_random(16, encryption_key);
+  for(int i = 0; i < 16; i++) {
+    encryption_key[i] %= 128;
+  }
+}
+
+static void bragi_set_device_keys(usbdevice* kb, unsigned char* pairing_id, unsigned char* encryption_key) {
+
+  uchar pkt[BRAGI_JUMBO_SIZE] = {0};
+
+  // Write the PAIRING_ID to the device first
+  bragi_open_handle(kb, BRAGI_GENERIC_HANDLE, BRAGI_RES_PAIRINGID);
+  memcpy(pkt + 7, pairing_id, sizeof(uchar) * 8);
+  bragi_write_to_handle(kb, pkt, BRAGI_GENERIC_HANDLE, sizeof(pkt), sizeof(uchar) * 8);
+  bragi_close_handle(kb, BRAGI_GENERIC_HANDLE);
+
+  // The ENCRYPTION_KEY must be writting directly after the PAIRING_ID
+  bragi_open_handle(kb, BRAGI_GENERIC_HANDLE, BRAGI_RES_ENCRYPTIONKEY);
+  memcpy(pkt + 7, encryption_key, sizeof(uchar) * 16);
+  bragi_write_to_handle(kb, pkt, BRAGI_GENERIC_HANDLE, sizeof(pkt), sizeof(uchar) * 16);
+  bragi_close_handle(kb, BRAGI_GENERIC_HANDLE);
+}
+
+void cmd_pair_bragi(usbdevice* kb, usbmode* mode, int dummy1, int dummy2, const char* to) {
+  (void)mode;
+  (void)dummy1;
+  (void)dummy2;
+
+  uchar encryption_key[16];
+  bragi_generate_encryption_key(encryption_key);
```

Comment:
Should probably check to see if the generation succeeded.

Reply:
Just adding a ckb_info call, or do you have some specific checks in mind to assure that this succeeded?